backend/src/main/kotlin/app/ehrenamtskarte/backend/application/webservice/dataloader/ApplicationDataLoader.kt

```diff
@@ -0,0 +1,64 @@
+package app.ehrenamtskarte.backend.application.webservice.dataloader
+
+import app.ehrenamtskarte.backend.application.database.ApplicationVerificationEntity
+import app.ehrenamtskarte.backend.application.database.ApplicationVerifications
+import app.ehrenamtskarte.backend.application.database.ApplicationVerifications.nullable
+import app.ehrenamtskarte.backend.application.database.Applications
+import app.ehrenamtskarte.backend.application.database.repos.ApplicationRepository
+import app.ehrenamtskarte.backend.application.webservice.schema.view.ApplicationVerificationView
+import app.ehrenamtskarte.backend.application.webservice.schema.view.ApplicationView
+import kotlinx.coroutines.runBlocking
+import org.dataloader.DataLoader
+import org.dataloader.DataLoaderFactory
+import org.jetbrains.exposed.sql.SortOrder
+import org.jetbrains.exposed.sql.select
+import org.jetbrains.exposed.sql.transactions.transaction
+import java.util.concurrent.CompletableFuture
+
+const val APPLICATION_LOADER_NAME = "APPLICATION_LOADER"
+
+val applicationLoader: DataLoader<Int, ApplicationView?> = DataLoaderFactory.newDataLoader { ids ->
+    CompletableFuture.supplyAsync {
+        runBlocking {
+            transaction {
+                ApplicationRepository.findByIds(ids).map { it?.let { ApplicationView.fromDbEntity(it) } }
+            }
+        }
+    }
+}
+
+const val VERIFICATIONS_BY_APPLICATION_LOADER_NAME = "VERIFICATIONS_BY_APPLICATION_LOADER"
+val verificationsByApplicationLoader: DataLoader<Int, List<ApplicationVerificationView>?> =
+    DataLoaderFactory.newDataLoader { ids ->
```

Comment:
> As [BatchLoader](https://javadoc.io/static/com.graphql-java/java-dataloader/2.2.3/org/dataloader/BatchLoader.html) batch functions are executed the resulting futures are cached using a cache implementation of choice, so they will only execute once. Individual cache keys can be cleared, so they will be re-fetched when referred to again.

We probably need to clear the cached result, when an application verification is accepted or rejected. When testing the icons by directly editing the database entry, the query will still return the old value until I restarted the backend service.

Reply:
> We probably need to clear the cached result, when an application verification is accepted or rejected. When testing the icons by directly editing the database entry, the query will still return the old value until I restarted the backend service.

Agreed. The question is: should we (A) disable the cache entirely for all/specific dataloaders, or should we (B) invalidate their caches for specific ids when we expect a change (the "faster" option)?
(A) is the "safer" option because we can't forget invalidating ids, but
(B) is the "faster" option, because we don't need to query unchanged data.

Reply:
I feel like (A) should be the right answer, as the UI should reflect the state of the DB at some point. In that regard I didn't find anything about the cache expiring after any time. But I am also unsure of how many Applications we might be talking about here. 🤔

Reply:
I created a different task for that (as it will affect more code than I want to touch this PR): https://github.com/digitalfabrik/entitlementcard/issues/851
We should nevertheless implement this before Release in Bayern.